### ISSUE TITLE
Tensorflow Debugger eats disk space with RNNs

### ISSUE THREAD

**Issue by alanhdu** [tensorflow/tensorflow#12465]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.3
- **Python version**:  3.5

### Describe the problem

I have an RNN that I'm trying to debug using `LocalCLIDebugWrapperSession` which runs on very long sequences (4800 steps). Even for a basic LSTM cell that I apply via `tf.nn.dynamic_rnn`, `tfdbg` uses a lot of disk space (>5 GB sometimes). When I accidentally collected gradient infromation via `tf.contrib.layers.optimize_loss(..., summaries=["gradients", ...])`, this ballooned even more to >70 GB (which promptly crashed my laptop).

From inspecting the `tfdbg` dump in `/tmp/`, it looks like this is because tfdbg` dumps out information for each time step. Especially with `tf.contrib.layers.optimize_loss` capturing gradients, this means that there are hundreds (thousands?) of small files being created on each time step.

In some sense, this is expected behavior (each time step represents a group of TF operations), but using 70 GB seems like a pretty sharp-edged API that's easy to mis-use. I'm not sure what to really do here -- maybe there's a way to somehow compress these files or to combine all these small files across time steps into one large file? It could also just be a documentation problem.

**Comment by caisq** [comment]
This possibility of filling large amount of disk space while using tfdbg to debug certain models is documented in this FAQ section: https://www.tensorflow.org/programmers_guide/debugger#frequently_asked_questions,

under the heading 
Q: The model I am debugging is very large. The data dumped by tfdbg fills up the free space of my disk. What can I do?

But I'm considering adding a configurable cap on how much disk space tfdbg consumes by default.

**Comment by alanhdu** [comment]
Huh... I somehow missed that FAQ section 😨.

One other problem with having hundreds of thousands of small files is that the tfdbg UI regularly freezes whenever I try to use it. Looking at strace, it looks like it's looping through all the files it created

```
...
newfstatat(14, "Sum_0_DebugIdentity_1503415213363935", {st_mode=S_IFREG|0664, st_size=15705, ...}, AT_SYMLINK_NOFOLLOW) = 0
unlinkat(14, "Sum_0_DebugIdentity_1503415213363935", 0) = 0
newfstatat(14, "Sum_1_0_DebugIdentity_1503415270902805", {st_mode=S_IFREG|0664, st_size=15709, ...}, AT_SYMLINK_NOFOLLOW) = 0
unlinkat(14, "Sum_1_0_DebugIdentity_1503415270902805", 0) = 0
...
```

**Comment by luchensk** [comment]
+1, found that it was very slow once turn on tfdbg dump by `DumpingDebugWrapperSession`, and I post the comparison info about my example training(300 steps), as below:

turn off tfdbg dump:
real	0m9.759s
user	0m8.315s
sys	0m1.625s

turn on tfdbg dump:
real	2m24.094s
user	2m19.819s
sys	0m12.707s

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 257 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 272 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 287 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 302 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 317 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 331 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 347 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 362 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 377 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
